specification.md

```diff
@@ -1294,6 +1315,19 @@ resources reserved through the advanced reservation represented by this
 ID.
 
 
+<a name="jobattributes-setcustomattribute"></a>
+```java
+void setCustomAttribute(String name, Object value);
+Object? getCustomAttribute(String name);
+```
+
+Allows setting/querying of custom attributes. Implementations should use a
+dictionary to implement the custom attributes. Additionally, implementations are
+encouraged to make sensible decisions on whether to store some or all of the
+fixed attributes in the same dictionary or not. It is, therefore, entirely
+possible for `getCustomAttribute("duration")` to return a value passed earlier
+to `setDuration()`, although the specific custom attribute name need not be
+`"duration"`.
```

Comment:
I'm a little bit concerned about violating the principle of least surprise if a user happens to use a key that they didn't realize was reserved/tied to a fixed attribute. It's admittedly a pretty low probability with the functions we have defined currently.  Thoughts on allowing `get`s of the fixed attributes but not `set`s via this API?

Reply:
I echo that concern actually.  I would think that we don't need to make these attributes writeable by the user of the API, but rather expect the implementation to provide read-only values.

Reply:
I would also suggest to return the custom attributes in one call.  Otherwise we would need to add inspection to inform the application about what attributes can be queries at that point (which may differ from one `JobStatus` object to the next).  So we could return a map, or a set of key/value pairs.  Well, I am a bit biased I guess -- see the version in #1 which addresses the same TODO a bit differently.

Reply:
Ignore my previous (deleted) comments for this method, I missed the point.  Let me instead please ask what the use case this method is trying to address?

Reply:
The assumption here is that users will not randomly set keys in this dictionary for no reason. And a user would have no reason to use a key unless an implementation assigned some function to that key.

Reply:
> Ignore my previous (deleted) comments for this method, I missed the point. Let me instead please ask what the use case this method is trying to address?

@andre-merzky I can't remember where this came from. I can't find it in the comments. @SteVwonder any ideas?

Reply:
I don't see anything in the google docs notes either.  I remember us mentioning that it is easy for people to abuse this functionality, but I can't remember what the specific intended use case was.

Reply:
I would not mind removing that method then until we are clear about the use case.

Reply:
Removed